rllib/utils/actor_manager.py

```diff
@@ -0,0 +1,324 @@
+from collections import defaultdict
+import logging
+from typing import Callable, List, Optional, Tuple, TypeVar
+
+import ray
+from ray.actor import ActorHandle
+from ray.exceptions import RayActorError
+from ray.util.annotations import DeveloperAPI
+
+
+logger = logging.getLogger(__name__)
+# Generic type var for foreach_* methods.
+T = TypeVar("T")
+
+
+class FaultTolerantActorManager:
+    """A manager that is aware of the healthiness states of remote actors.
+
+    Args:
+        actors: A list of ray remote actors to manage on. These actors must have an
+            ``apply`` method which takes a function with no parameters.
+        max_remote_requests_in_flight_per_actor: The maximum number of remote
+            requests that can be in flight per actor. Any requests made to the pool
+            that cannot be scheduled because the limit has been reached will be dropped.
+            This only applies to the async mode.
+    """
+
+    def __init__(
+        self,
+        actors: Optional[List[ActorHandle]] = None,
+        max_remote_requests_in_flight_per_actor: Optional[int] = 2,
+        ignore_worker_failure=True,
+    ):
+        self.__actors = actors or []
```

Comment:
Why two underscores?

Reply:
this is super private. so private inherited classes don't even have access.
by coding style, this means "absolutely nobody touches these variables/functions".

Reply:
In a subclass if you want to access a double underscored attribute python will yell at you.